packages/ui/src/views/user-management/components/dialogs/components/edit-user/edit-user.tsx

```diff
@@ -0,0 +1,106 @@
+import { useEffect } from 'react'
+import { SubmitHandler, useForm } from 'react-hook-form'
+
+import { Button, ButtonGroup, Dialog, Input } from '@/components'
+import { editUserSchema } from '@/views/user-management/components/dialogs/components/edit-user/schemas'
+import {
+  EditUserFormType,
+  IEditUserDialogProps
+} from '@/views/user-management/components/dialogs/components/edit-user/types'
+import { useStates } from '@/views/user-management/providers/state-provider'
+import { useUserManagementStore } from '@/views/user-management/providers/store-provider'
+import { zodResolver } from '@hookform/resolvers/zod'
+
+export function EditUserDialog({ handleUpdateUser, open, onClose }: IEditUserDialogProps) {
+  const { useTranslationStore, useAdminListUsersStore } = useUserManagementStore()
+
+  const { t } = useTranslationStore()
+  const { user } = useAdminListUsersStore()
+
+  const { loadingStates, errorStates } = useStates()
+  const { isUpdatingUser } = loadingStates
+  const { updateUserError } = errorStates
+
+  const {
+    register,
+    handleSubmit,
+    reset,
+    formState: { errors }
+  } = useForm<EditUserFormType>({
+    resolver: zodResolver(editUserSchema),
+    mode: 'onChange'
+  })
+
+  useEffect(() => {
+    if (user) {
+      reset({
+        userID: user.uid,
+        email: user.email,
+        displayName: user.display_name
+      })
+    }
+  }, [user, reset])
+
+  const onSubmit: SubmitHandler<EditUserFormType> = data => {
+    handleUpdateUser(data)
+  }
```

Comment:
Do we need this intermediate function? Could we just use `handleUpdateUser` directly? i.e.
```typescript jsx
<Button type="button" onClick={handleSubmit(handleUpdateUser)} disabled={isUpdatingUser}>
```